dask_sql/integrations/fugue.py

```diff
@@ -15,9 +17,24 @@
 
 from dask_sql.context import Context
 
-register_execution_engine(
-    "dask", lambda conf: DaskSQLExecutionEngine(conf), on_dup="overwrite"
-)
+
+@run_at_def
+def register_engines() -> None:
```

Comment:
Should we preface this function with an underscore to emphasize that it's not intended for external use?

```suggestion
def _register_engines() -> None:
```

Reply:
Sure, will change